src/tlo/util.py

```diff
@@ -484,6 +485,7 @@ def read_csv_files(folder: Path, files: Optional[list[str]] = None) -> DataFrame
     :py:func:`pandas.drop`.
 
     :param folder: Path to folder containing CSV files to read.
+    :param dtype: preferred datatype
```

Comment:
The `pandas.read_csv` method allows passing in a dictionary of datatypes (mapping from column names to datatypes) in cases where you want not just a single datatype for whole sheet but different datatypes per column. Are there any cases where we might need this functionality? As we passthrough the `dtype` argument straight to `pandas.read_csv` we could already pass in a dict here, but it might be worth documenting this is an additional option if we do think it will be used.

Reply:
Actually it seems like you do use explicitly use this ability to pass `dict` in test below so this should definitely be documented and typehint for `dtype` updated to `DtypeArg | dict[str, DtypeArg] | None`.